src/openrct2/drawing/Text.h

```diff
@@ -0,0 +1,56 @@
+#pragma region Copyright (c) 2014-2017 OpenRCT2 Developers
+/*****************************************************************************
+ * OpenRCT2, an open source clone of Roller Coaster Tycoon 2.
+ *
+ * OpenRCT2 is the work of many authors, a full list can be found in contributors.md
+ * For more information, visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * A full copy of the GNU General Public License can be found in licence.txt
+ *****************************************************************************/
+#pragma endregion
+
+#pragma once
+
+#include "../common.h"
+
+extern "C"
+{
+    #include "drawing.h"
+}
+
+enum class TextAlignment
+{
+    LEFT,
+    CENTRE,
+    RIGHT
+};
+
+struct TextPaint
+{
+    uint8           Colour;
+    sint16          SpriteBase;
+    bool            UnderlineText;
+    TextAlignment   Alignment;
+};
+
+class StaticLayout
+{
+private:
+    utf8string  _buffer;
```

Comment:
Default constructor should be hidden. I think copy constructor should be hidden too, there are only two instances so far that can easily be converted to non-`auto` variables and it doesn't look like this class is meant to be copied around.

Reply:
How to do these things?

Reply:
Just declare them in the private part of class declaration.

```cpp
class Static layout
{
private:
StaticLayout();
StaticLayout (const StaticLayout &);
```

Reply:
That causes this error to appear:

```
/Users/Marijn/Projects/OpenRCT2/Main/src/openrct2/drawing/Text.cpp:217:23: error: calling a private constructor of class 'StaticLayout'
        auto layout = StaticLayout(buffer, _legacyPaint, width);
                      ^
/Users/Marijn/Projects/OpenRCT2/Main/src/openrct2/drawing/Text.h:47:5: note: declared private here
    StaticLayout(const StaticLayout &);
```

Reply:
Yes, that's why I said you have to convert them to regular, non-`auto` variables that don't copy.

Reply:
Ah, my bad! Completely missed that.